simple/dense_directed_matrix.go

```diff
@@ -96,36 +143,45 @@ func (g *DirectedDenseGraph) To(n graph.Node) []graph.Node {
 			continue
 		}
 		if !isSame(g.mat.At(i, id), g.absent) {
-			neighbors = append(neighbors, Node(i))
+			neighbors = append(neighbors, g.Node(i))
 		}
 	}
 	return neighbors
 }
 
-func (g *DirectedDenseGraph) HasEdgeBetween(x, y graph.Node) bool {
+// HasEdgeBeteen returns whether an edge exists between nodes x and y without
+// considering direction.
+func (g *DirectedMatrix) HasEdgeBetween(x, y graph.Node) bool {
 	xid := x.ID()
 	yid := y.ID()
 	return xid != yid && (!isSame(g.mat.At(xid, yid), g.absent) || !isSame(g.mat.At(yid, xid), g.absent))
 }
 
-func (g *DirectedDenseGraph) Edge(u, v graph.Node) graph.Edge {
+// Edge returns the edge from u to v if such an edge exists and nil otherwise.
+// The node v must be directly reachable from u as defined by the From method.
+func (g *DirectedMatrix) Edge(u, v graph.Node) graph.Edge {
 	if g.HasEdgeFromTo(u, v) {
-		return Edge{F: u, T: v, W: g.mat.At(u.ID(), v.ID())}
+		return Edge{F: g.Node(u.ID()), T: g.Node(v.ID()), W: g.mat.At(u.ID(), v.ID())}
 	}
 	return nil
 }
 
-func (g *DirectedDenseGraph) HasEdgeFromTo(u, v graph.Node) bool {
+// HasEdgeFromTo returns whether an edge exists in the graph from u to v.
+func (g *DirectedMatrix) HasEdgeFromTo(u, v graph.Node) bool {
 	uid := u.ID()
 	vid := v.ID()
 	return uid != vid && !isSame(g.mat.At(uid, vid), g.absent)
 }
 
-func (g *DirectedDenseGraph) Weight(x, y graph.Node) (w float64, ok bool) {
+// Weight returns the weight for the edge between x and y if Edge(x, y) returns a non-nil Edge.
+// If x and y are the same node or there is no joining edge between the two nodes the weight
+// value returned is implementation dependent. Weight returns true if an edge exists between
```

Comment:
This seems to be copied from somewhere else. It is not implementation dependent, it is given by g.self and g.absent. Also in DirectedGraph and UndirectedGraph, if I am not mistaken.